### PR TITLE
Adds support for ES 7.7.0

### PR DESCRIPTION
**Pull request by ftianli-amzn**
*Issue #, if available:*
https://github.com/opendistro-for-elasticsearch/alerting/issues/204

*Description of changes:*

- Bump ODFE version to 1.8.0
- Bump Elasticsearch version to 7.7.0
- Bump JDK version to 14  
- Bump Gradle version to 6.4
- Bump nebula.ospackage version to 8.3.0
- Change the parameter of AlertingPlugin#createComponents (https://github.com/elastic/elasticsearch/pull/52604)
- Change the RestHandlers to use new way to register with the RestController (https://github.com/elastic/elasticsearch/pull/51950)
- Set the indices start with '.' to be hidden when creating to adapt with ES 7.x (https://github.com/elastic/elasticsearch/pull/51164)
- Fix a few broken tests because the wildcard expansion will not return hidden indices by default
- Update the requirement of JDK 14 in README and workflow files
- Update Release Notes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: alerting/src/test/kotlin/com/amazon/opendistroforelasticsearch/alerting/MonitorRunnerIT.kt [hunk 1]
```diff
@@ -65,7 +65,7 @@ class MonitorRunnerIT : AlertingRestTestCase() {
     fun `test execute monitor returns search result`() {
         val uniqueName = "unique name"
         val query = QueryBuilders.termQuery("monitor.name.keyword", uniqueName)
-        val input = SearchInput(indices = listOf("*"), query = SearchSourceBuilder().query(query))
+        val input = SearchInput(indices = listOf(".*"), query = SearchSourceBuilder().query(query))
```
Comment (dbbaughe) [inline alerting/src/test/kotlin/com/amazon/opendistroforelasticsearch/alerting/MonitorRunnerIT.kt:68]: Is this the way to search for hidden indices? `*` does not match but `.*` does match? Do you have a link to any docs on searching hidden indices?

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.2.2-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.4-bin.zip
```
Comment (dbbaughe) [inline gradle/wrapper/gradle-wrapper.properties:3]: I don't see any binary changes even though you increased gradle version?

File: .github/workflows/push-notification-jar.yml [hunk 3]
```diff
@@ -22,7 +22,7 @@ jobs:
       - name: Setup Java
         uses: actions/setup-java@v1
         with:
-          java-version: '13'
+          java-version: '14'
```
[the remark this answers is not in the document]
Comment (ftianli-amzn) [reply .github/workflows/push-notification-jar.yml:25]: Ah, seems I did a mistake, this workflow is not related with the plugin build process. I will change this back.

File: alerting/src/test/kotlin/com/amazon/opendistroforelasticsearch/alerting/MonitorRunnerIT.kt (hunk 1 above)
Comment (ftianli-amzn) [reply alerting/src/test/kotlin/com/amazon/opendistroforelasticsearch/alerting/MonitorRunnerIT.kt:68]: Here is a reference, in the PR of introducing the hidden index into Elasticsearch: https://github.com/elastic/elasticsearch/pull/51164,

> Wildcard expansion will not return hidden indices by default unless
the wildcard pattern begins with a .. This behavior is similar to
shell expansion of wildcards.

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (ftianli-amzn) [reply gradle/wrapper/gradle-wrapper.properties:3]: Thanks. I will use `gradle wrapper` to upgrade the Gradle version.

File: .github/workflows/push-notification-jar.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (dbbaughe) [reply .github/workflows/push-notification-jar.yml]: This workflow builds the plugin so it should be 14, you just need to update it in all places.

[the remark this answers is not in the document]
Comment (ftianli-amzn) [reply .github/workflows/push-notification-jar.yml]: Oh, sorry... Ok, I see. Thank you.

[the remark this answers is not in the document]
Comment (dbbaughe) [reply .github/workflows/push-notification-jar.yml:25]: I see a JAVA13_HOME below still, do we still need that or JAVA14_HOME?

[the remark this answers is not in the document]
Comment (ftianli-amzn) [reply .github/workflows/push-notification-jar.yml:25]: Got it. Totally there were 3 places with the java version "13", I have changed them all. :sweat_smile:

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (ftianli-amzn) [reply gradle/wrapper/gradle-wrapper.properties:3]: Have used `./gradlew wrapper --gradle-version 6.4` command to upgrade.

**Pull request merged**